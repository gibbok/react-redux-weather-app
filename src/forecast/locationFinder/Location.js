// @flow
import * as React from 'react'

/* eslint-disable no-undef */
type PropsType = {
  onLocationClick: Function,
  id: number,
  name: string,
  country:string
}
/* eslint-enable no-undef */

const Location:React.StatelessFunctionalComponent<any> = ({ onLocationClick, id, name, country }:PropsType):React.Element<any> => (
  <li>
    <a onClick={onLocationClick}>{name}, {country}</a>
  </li>
)

export default Location
