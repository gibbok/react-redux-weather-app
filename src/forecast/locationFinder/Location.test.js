import React from 'react'
import { render, mount, shallow } from 'enzyme'
import Location from './Location'
import sinon from 'sinon'
import toJson from 'enzyme-to-json'

describe('<Location />', () => {
  it('should render', () => {
    expect(shallow(
      <Location
        id={3067696}
        name='Prague'
        country='CZ'
        onLocationClick='test'
      />).contains(<li><a onClick='test'>Prague, CZ</a></li>)).toBe(true)
  })

  it('should render as snapshot', () => {
    const wrapper = shallow(
      <Location
        id={3067696}
        name='Prague'
        country='CZ'
        onLocationClick='test'
    />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should render to static HTML', function () {
    expect(render(
      <Location
        id={3067696}
        name='Prague'
        country='CZ'
        onLocationClick='test'
      />).text()).toEqual(`Prague, CZ`)
  })

  it('should mount in a full DOM', function () {
    expect(mount(
      <Location
        id={3067696}
        name='Prague'
        country='CZ'
        onLocationClick={() => {}}
      />
    ).find('li > a').length).toBe(1)
  })

  it('should simulates click events', () => {
    const onLinkClick = sinon.spy()
    const actualNode = shallow(
      <Location
        onLocationClick={onLinkClick}
      />
    )
    actualNode.find('a').simulate('click')
    sinon.assert.called(onLinkClick)
  })
})
