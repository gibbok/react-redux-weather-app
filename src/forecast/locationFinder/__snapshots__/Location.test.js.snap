// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Location /> should render as snapshot 1`] = `
<li>
  <a
    onClick="test"
  >
    Prague
    , 
    CZ
  </a>
</li>
`;
